hudi-client/hudi-java-client/src/test/java/org/apache/hudi/testutils/HoodieJavaClientTestHarness.java

```diff
@@ -429,22 +428,12 @@ private void runFullValidation(HoodieWriteConfig writeConfig,
       List<String> metadataTablePartitions = FSUtils.getAllPartitionPaths(
           engineContext, storage, HoodieTableMetadata.getMetadataTableBasePath(basePath), false);
 
-      List<MetadataPartitionType> enabledPartitionTypes = metadataWriter.getEnabledPartitionTypes();
-
-      assertEquals(enabledPartitionTypes.size(), metadataTablePartitions.size());
```

Comment:
For secondary and functional indexes, enabledPartitionTypes do not necessarily mean available partitions because secondary and functional index partition type can have multiple metadata partitions. This assertion does not add much value, hence removing it.